packages/browser/src/BacktraceBrowserRequestHandler.ts

```diff
@@ -0,0 +1,102 @@
+import {
+    BacktraceAttachment,
+    BacktraceReportSubmissionResult,
+    BacktraceRequestHandler,
+    ConnectionError,
+    DEFAULT_TIMEOUT,
+} from '@backtrace/sdk-core';
+import { BacktraceData } from '@backtrace/sdk-core/src/model/data/BacktraceData';
+
+export class BacktraceBrowserRequestHandler implements BacktraceRequestHandler {
+    private readonly UPLOAD_FILE_NAME = 'upload_file';
+    private readonly _timeout: number;
+    private readonly JSON_HEADERS = {
+        'Content-type': 'application/json',
+    };
+
+    constructor(
+        private readonly _options: {
+            url: string;
+            token?: string;
+            timeout?: number;
+        },
+    ) {
+        this._timeout = this._options.timeout ?? DEFAULT_TIMEOUT;
+    }
+    public async postError<T>(
+        submissionUrl: string,
+        data: BacktraceData,
+        attachments: BacktraceAttachment[],
+    ): Promise<BacktraceReportSubmissionResult<T>> {
+        const formData = this.createFormData(JSON.stringify(data), attachments);
+        return this.post(submissionUrl, formData);
+    }
+
+    public async post<T>(
+        submissionUrl: string,
+        payload: string | FormData,
+    ): Promise<BacktraceReportSubmissionResult<T>> {
+        const controller = new AbortController();
+        const id = setTimeout(() => controller.abort(), this._timeout);
+
+        try {
+            const response = await fetch(submissionUrl, {
+                method: 'POST',
+                body: payload,
+                headers: typeof payload === 'string' ? this.JSON_HEADERS : {},
+                signal: controller.signal,
+            });
+
+            clearInterval(id);
+
+            switch (response.status) {
+                case 200: {
+                    const result = response.json() as T;
```

Comment:
You should `await` this.

Reply:
done, thank you! I wonder why eslint didn't complain about that.